packages/loader/src/resource-deserialize/resources/scene/SceneParser.ts

```diff
@@ -1,36 +1,113 @@
-import { Engine, Entity, Scene } from "@oasis-engine/core";
+import { Engine, Entity, Loader, Scene } from "@oasis-engine/core";
 import { IEntity, IScene } from "../prefab/PrefabDesign";
 import { PrefabParser } from "../prefab/PrefabParser";
 import { ReflectionParser } from "../prefab/ReflectionParser";
+import { SceneParserContext } from "./SceneParserContext";
 
+/** @Internal */
 export class SceneParser {
+  /**
+   * Parse scene data.
+   * @param engine - the engine of the parser context
+   * @param sceneData - scene data which is exported by editor
+   * @returns a promise of scene
+   */
   static parse(engine: Engine, sceneData: IScene): Promise<Scene> {
     const scene = new Scene(engine);
-    const entitiesMap: Record<string, Entity> = {};
-    const entitiesConfigMap: Record<string, IEntity> = {};
-    const promises: Promise<Entity>[] = [];
-    const entitiesConfig = sceneData.entities;
-    for (const entity of entitiesConfig) {
-      entitiesConfigMap[entity.id] = entity;
-      promises.push(ReflectionParser.parseEntity(entity, engine));
-    }
+    const context = new SceneParserContext(sceneData, scene);
+    const parser = new SceneParser(context);
```

Comment:
`SceneParser` maybe should not create multi times,use static or just create once